packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java

```diff
@@ -428,7 +429,9 @@ private Query readFirestoreQuery(ByteBuffer buffer) {
 
       boolean isFilterQuery = parameters.containsKey("filters");
       if (isFilterQuery) {
-        Filter filter = filterFromJson((Map<String, Object>) parameters.get("filters"));
+        @SuppressWarnings("unchecked")
+        Filter filter =
+            filterFromJson((Map<String, Object>) Objects.requireNonNull(parameters.get("filters")));
```

Comment:
What would happen if `parameters.get("filters")` is null? Isn't this a different behaviour than the old approach?

Reply:
[Can never be null](https://github.com/firebase/flutterfire/pull/12577/files#diff-682b85d3ab07627cae8850bedbba3c3994dc2c361e1369178d2f8256d8fc64dfR430)